mbed/mbed.py

```diff
@@ -1324,15 +1325,28 @@ def get_env(self):
     def get_target(self, target=None):
         target_cfg = self.get_cfg('TARGET')
         target = target if target else target_cfg
+
+        if target and (target.lower() == 'detect' or target.lower() == 'auto'):
+            targets = self.get_detected_targets()
+            if targets == False:
+                error("The target detection requires that the 'mbed-ls' python module is installed.")
```

Comment:
It'd be good to add instructions on how to install mbed-ls here. Maybe something along these lines:

```
error("The target detection requires that the 'mbed-ls' python module is installed.\nYou can install mbed-ls by running 'pip install mbed-ls'.")
```